code/modules/shuttle/shuttle.dm

```diff
@@ -246,6 +245,28 @@
 		reserved_area = null
 	return ..()
 
+/obj/docking_port/stationary/picked
+	///Hods a list of map name strings for the port to pick from
+	var/list/shuttlekeys
+
+/obj/docking_port/stationary/picked/Initialize(mapload)
+	. = ..()
+	if(!LAZYLEN(shuttlekeys))
+		WARNING("Random docking port [id] loaded with no shuttle keys")
+		return
+	var/selectedid = pick(shuttlekeys)
+	roundstart_template = SSmapping.shuttle_templates[selectedid]
+
+/obj/docking_port/stationary/picked/whiteship
+	name = "Deep Space"
+	id = "whiteship_away"
+	dheight = 0
+	dir = 2
+	dwidth = 11
+	height = 22
+	width = 35
+	shuttlekeys = list("whiteship_meta", "whiteship_pubby", "whiteship_box", "whiteship_cere", "whiteship_kilo", "whiteship_donut", "whiteship_delta")
```

Comment:
What happens if the picked key doesn't actually exist? Say somebody removes one of these ships but doesn't know they need to update this var?

Reply:
It would have the exact same failcase as the json based system the whiteships use now

Reply:
I don't know what that failcase is, but good enough for me.